### DISCUSSION TITLE
Accessing the unique ID of the state node

### DISCUSSION THREAD

**Discussion by github0013** [statelyai/xstate#2479] [Q&A] [answered]
https://xstate.js.org/docs/guides/ids.html#relative-targets

https://github.com/statelyai/xstate/blob/6148d1c27d61380d94abae650be1b78ac03cb369/packages/core/src/types.ts#L620

I know this top level `id` is for the states, but are we able to access this id property from the service object?

```ts
const machine = createMachine({id: "myID"})
...
const service = ...
service.id // gives me a number, like 4.  Instead, I want to see "myID" 
```

If I wanted to identify which machine the service is interpreted from, is there a way?

**Comment by davidkpiano** [comment, marked as answer]
`service.machine.id` is what you need.

**Reply by github0013** [reply]
I was using `ActorRefFrom` to create a `service type` by referencing https://dev.to/mpocock1/how-to-manage-global-state-with-xstate-and-react-3if5  And it was why I didn't find anything useful to read the ID.

`interpret` and `useInterpret` both show these properties and methods below, including `machine`.

```ts
import { createMachine, ActorRefFrom, interpret } from "xstate"
const machine = createMachine({})
const service = interpret(machine)
```

![Screen_Shot_2021-07-30_at_14_38_16__2_](https://user-images.githubusercontent.com/809378/127606043-d0badb4e-0712-444b-82f6-c118dec5c2b0.png)



However, a service type created by `ActorRefFrom` only returns these.

```ts
type Service = ActorRefFrom<typeof machine>
const x: Service
```

![Screen Shot 2021-07-30 at 14 39 14](https://user-images.githubusercontent.com/809378/127606049-aeca194e-204b-422b-85f0-c7c82e59db36.png)

How do we properly create a service type for a machine?

**Reply by ChrisShank** [reply]
`InterpreterFrom<typeof machine>`!
